File: API/PageAndRDL/Stylesheets/C#/Stylesheets/StyleSheetsForm.cs
using System;
using System.IO;
using System.Windows.Forms;
using GrapeCity.ActiveReports.Document;
using GrapeCity.ActiveReports.PageReportModel;
using StyleSheet = GrapeCity.ActiveReports.SectionReportModel.StyleSheet;

namespace GrapeCity.ActiveReports.Samples.StyleSheets
{
	public partial class StyleSheetsForm : Form
	{
		private static string _externalStyleSheet = string.Empty;
		private const string RdlReport = "ReorderList.rdlx";
		private const string PageReport = "Delivery Slip.rdlx";
		private static readonly string OutputFolder = Application.StartupPath;
		private static readonly string ReportsPath = Path.GetFullPath(Path.Combine(OutputFolder,@"..\..\..\..\Reports"));
		private FileInfo _file;

		public StyleSheetsForm()
		{
			InitializeComponent();
		}

		private void buttonRunReport_Click(object sender, EventArgs e)
		{
			PageReport report = null;
			
			// Select the report in the Viewer.
			// 
			if (radioButtonRDLReport.Checked)
				_file = new FileInfo(Path.Combine(ReportsPath, RdlReport));
			else if (radioButtonPageReport.Checked)
				_file = new FileInfo(Path.Combine(ReportsPath , PageReport));
			report = new PageReport(_file);

			if (radioButtonEmbeddedStyle.Checked)
			{
				report.Report.StyleSheetSource = StyleSheetSource.Embedded;
				report.Report.StyleSheetValue = "BaseStyle";
			}
			else
			{
				report.Report.StyleSheetSource = StyleSheetSource.External;
			} 
			if (radioButtonOrangeStyle.Checked)
			{
				report.Report.StyleSheetValue = Path.Combine(ReportsPath, "ModernStyle.rdlx-styles");
			}
			if (radioButtonGreenStyle.Checked)
			{
				report.Report.StyleSheetValue = Path.Combine(ReportsPath, "FaxSheetStyle.rdlx-styles");
			}
			if (radioButtonBlueStyle.Checked)
			{
				report.Report.StyleSheetValue = Path.Combine(ReportsPath, "HighContrastStyle.rdlx-styles");
			}
			if (radioButtonCustomStyle.Checked)
			{
				report.Report.StyleSheetValue = Path.Combine(ReportsPath, _externalStyleSheet);
			}

			var pageDocument = new PageDocument(report);
			reportViewer.LoadDocument(pageDocument);
		}

		private void buttonChooseExtStyle_Click(object sender, EventArgs e)
		{
			// Select the external stylesheet to apply on the report.
			// 
			FileDialog openFileDialog = new OpenFileDialog();
			openFileDialog.Filter = Properties.Resources.FilterText;
			openFileDialog.InitialDirectory = ReportsPath;
			openFileDialog.CheckFileExists = true;

			if (openFileDialog.ShowDialog(this) == DialogResult.OK)
			{
				FileInfo styleSheetFile = new FileInfo(openFileDialog.FileName);
				_externalStyleSheet = styleSheetFile.FullName;
				radioButtonCustomStyle.Text = Properties.Resources.StyleSheetText + styleSheetFile.Name;
				radioButtonCustomStyle.AutoEllipsis = true;
				radioButtonCustomStyle.Checked = true;
			}
		}
	}
}

File: API/PageAndRDL/Stylesheets/VB.NET/Stylesheets/StyleSheetsForm.vb
Imports System.IO
Imports GrapeCity.ActiveReports.Document
Imports GrapeCity.ActiveReports.PageReportModel

Public Class StyleSheetsForm
	Private _externalStyleSheet As String = String.Empty
	Private Const RdlReport As String = "ReorderList.rdlx"
	Private Const PageReport As String = "Delivery Slip.rdlx"
	Private ReadOnly _outputFolder As String = Application.StartupPath
	Private _reportsPath As String = Path.GetFullPath(Path.Combine(_outputFolder, "..\..\..\..\Reports"))
	Private _file As FileInfo

	Private Sub buttonChooseExtStyle_Click(ByVal sender As System.Object, ByVal e As EventArgs) Handles buttonChooseExtStyle.Click

		'Select an external stylsheet to apply.
		'

		Dim openFileDialog As FileDialog = New OpenFileDialog()
		openFileDialog.Filter = My.Resources.FilterText
		openFileDialog.InitialDirectory = _reportsPath
		openFileDialog.CheckFileExists = True

		If openFileDialog.ShowDialog(Me) = DialogResult.OK Then
			Dim styleSheetFile As FileInfo = New FileInfo(openFileDialog.FileName)
			_externalStyleSheet = styleSheetFile.FullName
			radioButtonCustomStyle.Text = My.Resources.StyleSheetText + styleSheetFile.Name
			radioButtonCustomStyle.AutoEllipsis = True
			radioButtonCustomStyle.Checked = True
		End If
	End Sub

	Private Sub buttonRunReport_Click(ByVal sender As System.Object, ByVal e As EventArgs) Handles buttonRunReport.Click
		Dim report As PageReport

		'Select the report in the viewer.
		If radioButtonRDLReport.Checked Then
			_file = New FileInfo(Path.Combine(_reportsPath, RdlReport))
		ElseIf radioButtonPageReport.Checked Then
			_file = New FileInfo(Path.Combine(_reportsPath, PageReport))
		End If
		report = New PageReport(_file)

		If radioButtonRDLReport.Checked Then
			_file = New FileInfo(Path.Combine(_reportsPath, RdlReport))
		End If

		If radioButtonPageReport.Checked Then
			_file = New FileInfo(Path.Combine(_reportsPath, PageReport))
		End If

		If radioButtonEmbeddedStyle.Checked Then
			report.Report.StyleSheetSource = StyleSheetSource.Embedded
			report.Report.StyleSheetValue = "BaseStyle"
		Else
			report.Report.StyleSheetSource = StyleSheetSource.External
		End If

		If radioButtonOrangeStyle.Checked Then
			report.Report.StyleSheetValue = Path.Combine(_reportsPath, "ModernStyle.rdlx-styles")
		End If

		If radioButtonGreenStyle.Checked Then
			report.Report.StyleSheetValue = Path.Combine(_reportsPath, "FaxSheetStyle.rdlx-styles")
		End If

		If radioButtonBlueStyle.Checked Then
			report.Report.StyleSheetValue = Path.Combine(_reportsPath, "HighContrastStyle.rdlx-styles")
		End If

		If radioButtonCustomStyle.Checked Then
			report.Report.StyleSheetValue = Path.Combine(_reportsPath, _externalStyleSheet)
		End If

		Dim pageDocument As PageDocument = New PageDocument(report)
		reportViewer.LoadDocument(pageDocument)
	End Sub

End Class
